src/net/client/mod.rs

```diff
@@ -1,8 +1,29 @@
+#![cfg_attr(
+    not(feature = "unstable-client-transport"),
+    doc = " The `unstable-client-transport` feature is necessary to enable this module."
+)]
 //! Sending requests and receiving responses.
 //!
 //! This module provides DNS transport protocols that allow sending a DNS
 //! request and receiving the corresponding reply.
 //!
+//! Currently the following transport protocols are supported:
+//! * [dgram] DNS over a datagram protocol, typically UDP.
+//! * [stream] DNS over an octet stream protocol, typically TCP or TLS.
+//!   Only a single connection is supported.
+//!   The transport works as long as the connection continues to exist.
+//! * [multi_stream] This is a layer on top [stream] where new connections
+//!   are established as old connections are closed (or fail).
+//! * [dgram_stream] This is a combination of [dgram] and [multi_stream].
+//!   This is typically needed because a request over UDP can receive
+//!   a truncated response, which should be retried over TCP.
+//! * [redundant] This transport multiplexes requests over a collection of
+//!   transport connections. The [redundant] transport favors the connection
+//!   with the lowest response time. And of the other transports can be added
+//!   as upstream transports.
```

Comment:
I don't understand what "And of the other transports can be added as upstream transports" means, the English doesn't "parse" for me.